### PR TITLE
issue 2269: follow up: add unit test for the negative scenarios

### PR DESCRIPTION
**Pull request by beiwei30**
# Description

follow up action for #2268: add unit test for the negative scenarios

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #2269 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Unit tests passing
* [x] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pkg/grpc/api_test.go [hunk 1]
```diff
@@ -645,14 +657,15 @@ func TestGetSecret(t *testing.T) {
...
-			AllowedSecrets: []string{"good-key"},
+			AllowedSecrets: []string{"error-key", "good-key"},
 		},
 	}
 	expectedResponse := "life is good"
 	storeName := "store1"
 	deniedStoreName := "store2"
 	restrictedStore := "store3"
 	unrestrictedStore := "store4" // No configuration defined for the store
+	nonExistingStore := "store5"  // Non-existing store
```
Comment (tcnghia) [inline pkg/grpc/api_test.go:668]:
```suggestion
	nonExistingStore := "nonexistent"  // Non-existing store
```
These already have descriptive names, but I would also use descriptive values for them. Sometimes only the values appears in the log and could be helpful when debugging as well.

File: pkg/grpc/api_test.go [hunk 2]
```diff
@@ -739,68 +768,274 @@ func TestGetSecret(t *testing.T) {
...
+	assert.Equal(t, codes.FailedPrecondition, status.Code(err))
+}
+
+func TestSaveState(t *testing.T) {
+	fakeStore := &daprt.MockStateStore{}
+	fakeStore.On("BulkSet", mock.MatchedBy(func(reqs []state.SetRequest) bool {
+		if len(reqs) == 0 {
+			return false
+		}
+		return reqs[0].Key == "fakeAPI||key1"
```
Comment (tcnghia) [inline pkg/grpc/api_test.go:796]: can we please change `key1`, `key2` to be something more descriptive so it's easier to connect when reading the test cases below?

**Review by tcnghia — CHANGES_REQUESTED** [review]

File: pkg/grpc/api_test.go (hunk 2 above)
Comment (beiwei30) [reply pkg/grpc/api_test.go]: fixed.

**Comment by beiwei30** [comment]
@tcnghia would you mind to take a look again?

**Review by tcnghia — APPROVED** [review]
This looks good to me. Thanks for improving unit tests.
